### DISCUSSION TITLE
Adding a random function to factories

### DISCUSSION THREAD

**Discussion by chrislurty** [laravel/framework#36289] [Ideas]
Is there any interest in adding a random function to factories?
` User::factory()->rand(0, 3)->create(); `

I can create a PR if this is approved.

The use case would be for creating related models with a factory.
Passing rand() into the count method generates the same number of items for each,
where this would allow for a random number of items.

Same number of post for every user :
`
$user = User::factory()->count(3)->has(Post::factory()->count(rand(0, 3)))->create();
`

Random number of posts for each user :
`
$user = User::factory()->count(3)->has(Post::factory()->rand(0, 3))->create();
`

Possible implementation :

/**
     * Specify a random number of models to be generated.
     *
     * @param  int  $min
     * @param  int  $max
     * @return static
     */
    public function rand(int $min, int $max)
    {
        return $this->newInstance(['count' => rand($min, $max)]);
    }

**Comment by browner12** [comment]
I've actually been trying to take randomness **out** of my tests. Randomness *could* give a fall sense of security if your tests are green, but never actually hitting the random event that causes them to fail.  IMO, if you truly need to test a condition, be explicit and test it.

And there's nothing more frustrating than a test that passes on some runs, and fails on others.

I've recently just removed all the `->optional()` calls from my factories.

I'd almost be more interested in being able to explicitly make one User with 0 posts, one User with 1 post, and one User with 2 posts.

**Reply by chrislurty** [reply]
I agree that feature tests probably should avoid random data.

This would be more useful  for scalability/performance testing.
 As well as generating database records for our development environment.

**Reply by matsn0w** [reply]
I agree with both of you, it would be nice if you'd at least have to option to do it.

**Comment by derpoho** [comment]
Late to the party.

Pull Request ist open #49637. This solves this.
